File: commonutils/src/main/java/net/maxsmr/commonutils/logger/LogcatLogger.java
package net.maxsmr.commonutils.logger;

import org.jetbrains.annotations.Nullable;
import android.util.Log;

public class LogcatLogger extends BaseTagLogger {

    public LogcatLogger(@Nullable String tag) {
        super(tag);
    }

    @Override
    public void v(String message) {
        if (isLoggingEnabled()) {
            if (message != null) {
                Log.v(tag, message);
            }
        }
    }

    @Override
    public void v(Throwable exception) {
        if (isLoggingEnabled()) {
            if (exception != null) {
                Log.v(tag, exception.getMessage(), exception);
            }
        }
    }

    @Override
    public void v(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            if (message != null && exception != null) {
                Log.v(tag, message, exception);
            } else if (message != null) {
                v(message);
            } else if (exception != null) {
                v(exception);
            }
        }
    }

    @Override
    public void d(String message) {
        if (isLoggingEnabled()) {
            if (message != null) {
                Log.d(tag, message);
            }
        }
    }

    @Override
    public void d(Throwable exception) {
        if (isLoggingEnabled()) {
            if (exception != null) {
                Log.d(tag, exception.getMessage(), exception);
            }
        }
    }

    @Override
    public void d(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            if (message != null && exception != null) {
                Log.d(tag, message, exception);
            } else if (message != null) {
                d(message);
            } else if (exception != null) {
                d(exception);
            }
        }
    }

    @Override
    public void i(String message) {
        if (isLoggingEnabled()) {
            if (message != null) {
                Log.i(tag, message);
            }
        }
    }

    @Override
    public void i(Throwable exception) {
        if (isLoggingEnabled()) {
            if (exception != null) {
                Log.i(tag, exception.getMessage(), exception);
            }
        }
    }

    @Override
    public void i(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            if (message != null && exception != null) {
                Log.i(tag, message, exception);
            } else if (message != null) {
                i(message);
            } else if (exception != null) {
                i(exception);
            }
        }
    }

    @Override
    public void w(String message) {
        if (isLoggingEnabled()) {
            if (message != null) {
                Log.w(tag, message);
            }
        }
    }

    @Override
    public void w(Throwable exception) {
        if (isLoggingEnabled()) {
            if (exception != null) {
                Log.w(tag, exception.getMessage(), exception);
            }
        }
    }

    @Override
    public void w(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            if (message != null && exception != null) {
                Log.w(tag, message, exception);
            } else if (message != null) {
                w(message);
            } else if (exception != null) {
                w(exception);
            }
        }
    }

    @Override
    public void e(String message) {
        if (isLoggingEnabled()) {
            if (message != null) {
                Log.e(tag, message);
            }
        }
    }

    @Override
    public void e(Throwable exception) {
        if (isLoggingEnabled()) {
            if (exception != null) {
                Log.e(tag, exception.getMessage(), exception);
            }
        }
    }

    @Override
    public void e(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            if (message != null && exception != null) {
                Log.e(tag, message, exception);
            } else if (message != null) {
                e(message);
            } else if (exception != null) {
                e(exception);
            }
        }
    }

    @Override
    public void wtf(String message) {
        if (isLoggingEnabled()) {
            if (message != null) {
                Log.wtf(tag, message);
            }
        }
    }

    @Override
    public void wtf(Throwable exception) {
        if (isLoggingEnabled()) {
            if (exception != null) {
                Log.wtf(tag, exception.getMessage(), exception);
            }
        }
    }

    @Override
    public void wtf(String message, Throwable exception) {
        if (isLoggingEnabled()) {
            if (message != null && exception != null) {
                Log.wtf(tag, message, exception);
            } else if (message != null) {
                wtf(message);
            } else if (exception != null) {
                wtf(exception);
            }
        }
    }
}

File: commonutils/src/main/kotlin/net/maxsmr/commonutils/logger/LogcatLogger.kt
package net.maxsmr.commonutils.logger

import android.util.Log

class LogcatLogger(tag: String) : BaseLogger(tag) {

    override fun _v(message: String) {
        Log.v(tag, message)
    }

    override fun _v(exception: Throwable) {
        Log.v(tag, exception.message, exception)
    }

    override fun _v(message: String, exception: Throwable?) {
        Log.v(tag, message, exception)
    }

    override fun _d(message: String) {
        Log.d(tag, message)
    }

    override fun _d(exception: Throwable) {
        Log.d(tag, exception.message, exception)
    }

    override fun _d(message: String, exception: Throwable?) {
        Log.d(tag, message, exception)
    }

    override fun _i(message: String) {
        Log.i(tag, message)
    }

    override fun _i(exception: Throwable) {
        Log.i(tag, exception.message, exception)
    }

    override fun _i(message: String, exception: Throwable?) {
        Log.i(tag, message, exception)
    }

    override fun _w(message: String) {
        Log.w(tag, message)
    }

    override fun _w(exception: Throwable) {
        Log.w(tag, exception.message, exception)
    }

    override fun _w(message: String, exception: Throwable?) {
        Log.w(tag, message, exception)
    }

    override fun _e(message: String) {
        Log.e(tag, message)
    }

    override fun _e(exception: Throwable) {
        Log.e(tag, exception.message, exception)
    }

    override fun _e(message: String, exception: Throwable?) {
        Log.e(tag, message, exception)
    }

    override fun _wtf(message: String) {
        Log.wtf(tag, message)
    }

    override fun _wtf(exception: Throwable) {
        Log.wtf(tag, exception.message, exception)
    }

    override fun _wtf(message: String, exception: Throwable?) {
        Log.wtf(tag, message, exception)
    }
}
